### PR TITLE
Update Matic.js doc - adding "Message" doc 

### PR DESCRIPTION
**Pull request by EmpieichO**
Update Matic.js doc - adding "Message" doc  ... as per issue https://polygon.atlassian.net/browse/DOCS-103

### CODE REVIEW

File: docs/tools/matic-js/zkevm/message.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
+This document entails customization of wrapped tokens using adapter contracts, transfer of tokens between Ethereum and Polygon zkEVM networks, and how Matic.js can be used to bridge assets from Ethereum to Polygon zkEVM and vice versa.
+
+## Basic functions for error passing
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md:3]: This section should go at the bottom of the doc, perhaps under a separate heading #Debugging

File: docs/tools/matic-js/zkevm/message.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+We refer to Ethereum as the _root_ chain and zkEVM as the _child_ chain.
+
+Below we provide the two basic functions used for _error passing_ in each of the two directions: L1 --> L2 and L2 --> L1.
+
+**Message**
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md:9]: What does **Message** refer to?

File: docs/tools/matic-js/zkevm/message.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+- For Root to child
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
- From root to child
```

File: docs/tools/matic-js/zkevm/message.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        destinationAddress: string,
+        amount: TYPE_AMOUNT,
+        metadata: string,
+        option: ITransactionOption
+);
+
+// proof can be found from the proof gen API
+```
+
+- For child to root
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
- From child to root
```

File: docs/tools/matic-js/zkevm/message.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+// proof can be found from the proof gen API
+```
+
+
+## Bridging customized ERC20 token
+
+The existing zkEVM bridge uses the ERC-20 standard contract for creating wrapped tokens depending on the token's native network.
+
+At often times organizations want to customise their wrapped tokens by extending some functionalities.
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
Often, organizations want to customise their wrapped tokens by extending some functionalities. 
```

File: docs/tools/matic-js/zkevm/message.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+```
+
+
+## Bridging customized ERC20 token
+
+The existing zkEVM bridge uses the ERC-20 standard contract for creating wrapped tokens depending on the token's native network.
+
+At often times organizations want to customise their wrapped tokens by extending some functionalities. 
+
+For instance, these functionalities could include: blacklisting, putting a cap on minting, or any sound auxiliary functionality.
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
These functionalities could include: blacklisting, putting a cap on minting, or any sound auxiliary functionality.
```

File: docs/tools/matic-js/zkevm/message.md [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+
+This can be done by deploying adapter contracts that use the messaging layer of the bridge.
+
+However, for the sake of maintaining consistency among wrapped tokens, strict standards must be followed when extending their functionalities.
+
+
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
```

File: docs/tools/matic-js/zkevm/message.md [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+## Standardizations
+
+
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
```

File: docs/tools/matic-js/zkevm/message.md [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+1. Adapter contracts need to implement the [Polygon bridge library](https://github.com/0xPolygonHermez/code-examples/blob/main/customERC20-bridge-example/contracts/base/PolygonERC20BridgeBase.sol) and expose `bridgeToken()` and `onMessageReceived()` functions.
+2. There should be two separate adapter contracts; `OriginChainBridgeAdapter` and `WrapperChainBridgeAdapter`.
+3. `bridgeToken()` function should match the exact function signature and be similar to [this](https://github.com/maticnetwork/static/blob/master/network/mainnet/cherry/artifacts/zkevm/ZkEVMBridgeAdapter.json) ABI.
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]: How is the function similar to an ABI?

File: docs/tools/matic-js/zkevm/message.md [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+
+**Good to have**:
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
### Nice to have
```

Comment (kmurphypolygon) [reply docs/tools/matic-js/zkevm/message.md]: Better to use headers than bold

File: docs/tools/matic-js/zkevm/message.md [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+        "wrappedTokenNetwork": 1, // 1 here is networkId of zkEvm,
+        "extensions": {
+            "rootAddress": "Ethereum Address of the Token",
+            "wrapperChainBridgeAdapter": "",
+            "originChainBridgeAdapter": "",
+            }
+    },
+    ```
+
+
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
```

File: docs/tools/matic-js/zkevm/message.md [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+    Note: If the token is ethereum native then `originTokenNetwork` should be `0`. If the token is zkEVM native, then `originTokenNetwork` should be `1`. Similarly for the `wrappedTokenNetwork`.
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md:196]: Can you make this an admonition so it renders nicely?

File: docs/tools/matic-js/zkevm/message.md [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
```

File: mkdocs.yml [hunk 14]
```diff
@@ -455,6 +455,7 @@ nav:
           - zkEVM:
               - zkEVM client:  tools/matic-js/zkevm/initialize.md
               - ERC20:  tools/matic-js/zkevm/erc20.md
+              - Message: tools/matic-js/zkevm/message.md
```
Comment (kmurphypolygon) [inline mkdocs.yml:458]: This is the wrong place for this document. It needs to be in the Matic.js section of Developer tools and I'm not sure the title is correct.

File: docs/tools/matic-js/zkevm/message.md [hunk 15]
```diff
@@ -0,0 +1,259 @@
+This document entails customization of wrapped tokens using adapter contracts, transfer of tokens between Ethereum and Polygon zkEVM networks, and how Matic.js can be used to bridge assets from Ethereum to Polygon zkEVM and vice versa.
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md]:
```suggestion
This document shows you how to customize wrapped tokens using adapter contracts, transfer tokens between Ethereum and Polygon zkEVM networks, and how to use Matic.js to bridge assets from Ethereum to Polygon zkEVM and vice versa.
```

File: docs/tools/matic-js/zkevm/message.md [hunk 16]
```diff
@@ -0,0 +1,259 @@
...
+        destinationNetwork: number,
+        destinationAddress: string,
+        amount: TYPE_AMOUNT,
+        metadata: string,
+        option: ITransactionOption
+);
+
+// proof can be found from the proof gen API
+```
+
```
Comment (kmurphypolygon) [inline docs/tools/matic-js/zkevm/message.md:68]:
```suggestion
```

File: docs/tools/matic-js/zkevm/message.md (hunk 15 above)
Comment (kmurphypolygon) [reply docs/tools/matic-js/zkevm/message.md]: Ideally, given this first para, the content should be split like this, and could be separate pages, to match with the same/similar headings:

Section 1: customizing wrapped tokens with adapter contract.
Section 2: transfer tokens between ..
Section 3: use Matic to bridge assets ... (this section seems to come first or it is badly named further down)

**Review by kmurphypolygon — APPROVED** [review]

**Comment by kmurphypolygon** [comment]
The doc name is not clear either. **How to send messages** sounds better, but is that what it's about? I'm not sure @EmpieichO

File: docs/tools/matic-js/zkevm/message.md (hunk 2 above)
Comment (EmpieichO) [reply docs/tools/matic-js/zkevm/message.md:9]: I realise that I should have removed "**Message**". I will do that.

**Pull request merged**